Unit Tests/Tests/Redux/Reducers/ConcreteReductionManagerTests.swift

```diff
@@ -0,0 +1,207 @@
+import XCTest
+import TestUtilities
+@testable import PusherChatkit
+
+class ConcreteReductionManagerTests: XCTestCase {
+    
+    // MARK: - Tests
+    
+    func test_reduce_withReceivedInitialStateAction_returnsStateFromDedicatedReducer() {
+        
+        /******************/
+        /*---- GIVEN -----*/
+        /******************/
+        
+        let expectedState = ChatState(
+            currentUser: .populated(
+                identifier: "alice",
+                name: "Alice A"
+            ),
+            joinedRooms: .empty
+        )
+        
+        let userSubscriptionInitialStateReducer = StubReducer<ReceivedInitialStateAction, ChatState>(reducer_expectedState: expectedState, reducer_expectedCallCount: 1)
+        let userSubscriptionRemovedFromRoomReducer = DummyReducer<ReceivedRemovedFromRoomAction, ChatState>()
+        
+        let sut = ConcreteReductionManager(userSubscriptionInitialStateReducer: userSubscriptionInitialStateReducer.reducer,
+                                           userSubscriptionRemovedFromRoomReducer: userSubscriptionRemovedFromRoomReducer.reducer)
+        
+        let currentState: ChatState = .empty
+        
+        let action = ReceivedInitialStateAction(
+            event: Wire.Event.InitialState(
+                currentUser: Wire.User(
+                    identifier: "alice",
+                    name: "Alice A",
+                    avatarURL: nil,
+                    customData: nil,
+                    createdAt: Date.distantPast,
+                    updatedAt: Date.distantFuture,
+                    deletedAt: nil
+                ),
+                rooms: [],
+                readStates: [],
+                memberships: []
+            )
+        )
+        
+        /******************/
+        /*----- WHEN -----*/
+        /******************/
+        
+        let result = sut.reduce(action: action, state: currentState)
```

Comment:
Is `newState` or `updatedState` a better variable name than `result`?  (this applies to all your tests).

Reply:
Actually I've changed my mind 🤣 .  I think `inputState` and `outputState` might be nicer.   So in the of this case of this test you'd rename as follows:

`expectedState` -> `initialStateReducer_inputState`
`currentState` -> `inputState`
`result` -> `outputState`

and probably add `let expectedState = initialStateReducer_inputState` at the top of the "THEN" section.

Reply:
Done